cartofacade/capi.go

```diff
@@ -43,9 +43,24 @@ type CartoLibInterface interface {
 	Terminate() error
 }
 
+// SlamMode represents the lidar configuration
+type SlamMode int64
+
+const (
+	// UnknownMode denotes an unknown slam mode
+	UnknownMode SlamMode = iota
+	// MappingMode denotes the slam algo is in mapping mode
+	MappingMode
+	// LocalizingMode denotes the slam algo is in localizing only mode
+	LocalizingMode
+	// UpdatingMode denotes the slam algo is in updating mode
+	UpdatingMode
+)
+
 // Carto holds the c type viam_carto
 type Carto struct {
 	value *C.viam_carto
+	SlamMode
```

Comment:
Adding the slam mode so that integration tests can assert on what mode the slam algo is configured to, which was previously derived through logs.